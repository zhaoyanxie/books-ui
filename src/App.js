import React, { Component } from 'react';
import Author from './Author';
import Book from './Book';
import './App.css';

class App extends Component {
	constructor() {
		super();
		this.state = {
			books: [],
			authors: []
		};
	}

	render() {
		return (
			<div>
				<Book />
				<Author />
			</div>
		);
	}
}

export default App;
