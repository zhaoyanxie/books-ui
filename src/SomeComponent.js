import React from 'react';

const SomeComponent = () => {
	return <div>I am from SomeComponent</div>;
};

module.exports = SomeComponent;
