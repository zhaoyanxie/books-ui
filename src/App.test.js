import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from './App';
import SomeComponent from './SomeComponent';

it('renders App and test snapshot', () => {
	const renderer = new ShallowRenderer();
	renderer.render(<App />);
	const output = renderer.getRenderOutput();

	expect(output).toMatchSnapshot();
});

it('renders Some Component', () => {
	const renderer = new ShallowRenderer();
	renderer.render(<SomeComponent />);
	const output = renderer.getRenderOutput();

	expect(output.type).toBe('div');
	expect(output.props.children).toEqual('I am from SomeComponent');
	expect(output).toMatchSnapshot();
});
