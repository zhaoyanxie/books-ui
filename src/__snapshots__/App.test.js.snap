// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App and test snapshot 1`] = `
<div>
  <Book />
  <h1>
    AUTHORS
  </h1>
  <Author />
</div>
`;

exports[`renders Some Component 1`] = `
<div>
  I am from SomeComponent
</div>
`;
